frontend/src/store/history/reducer.ts

```diff
@@ -33,8 +33,11 @@ const reducer = createReducer(initialState, (builder) => {
 
   builder.addCase(getUserVideoHistoryRecord.fulfilled, (state, { payload }) => {
     let newList: HistoryListType[] = state.data.list;
+    const { currentPage } = payload;
 
-    if (!isObjectUniqueIdContainInTwoArray(newList, payload.list)) {
+    if (currentPage === 1) {
+      newList = payload.list;
```

Comment:
you can get `payload.list` with destructuring `const { currentPage, list } = payload;`